### PR TITLE
feat (Catalog): Manage Controls Hierarchy

### PR DESCRIPTION
**Pull request by mpemy**

### CODE REVIEW

**Comment by mpemy** [comment]
@Bronstrom 
This PR is ready for review. Below are some screenshots
image
image
This how you can run this branch.
1.  In file App.js
set: const isDevMode = true;
2. In file OSCALLoader
 set const isDevMode = true;
3. To build the frontend run
  npm run build
4. To view the frontend run
 ./scripts/watch-viewer
5. To run the back.
checkout the branch https://github.com/EasyDynamics/oscal-prototyping/tree/moustapha/branch
and run it.

Let me know if you have any question

**Review by Bronstrom — COMMENTED** [review]
Looking good so far @mpemy! I really appreciate the instructions and screenshots.

I've attempted to add a new control, but it seems like the `list_all_controls` transactions fails, and then a second later, I'll get a success message of the control being added: 

image

It almost seems like the control isn't being added in time for the `list_all_controls` call. Any clue on why this is occurring? I'm using the latest version of OTSL and rebuilt the front-end before running `./scripts/watch-viewer`.

I've also provided some code comments below.

File: packages/oscal-react-library/src/components/OSCALCatalogBaseline.tsx [hunk 1]
```diff
@@ -520,20 +467,653 @@ export default function OSCALCatalogBaseline() {
...
+                    <UploadTypo
+                      variant="h6"
+                      sx={{
+                        top: 250,
+                        left: 490,
+                        width: 100,
+                        position: "absolute",
+                      }}
+                    >
+                      OR
```
[the remark this answers is not in the document]
Comment (mpemy) [reply packages/oscal-react-library/src/components/OSCALCatalogBaseline.tsx]: Thanks @Bronstrom, I addressed this concern.

File: packages/oscal-react-library/src/components/OSCALCatalogManageGroup.tsx [hunk 2]
```diff
@@ -0,0 +1,986 @@
...
+      <Container
+        onClick={handleNewGroupClick}
+        sx={{ height: 73, border: "1px solid #D2D2D2", background: "#EAEAEA", width: 312 }}
+      >
+        <Container
+          sx={{
+            height: 73,
+            left: 20,
+            position: "absolute",
+            background: "#F6F6F6",
```
[the remark this answers is not in the document]
Comment (mpemy) [reply packages/oscal-react-library/src/components/OSCALCatalogManageGroup.tsx]: Done

**Review by Bronstrom — APPROVED** [review]
This looks good to me @mpemy! Thanks for addressing the comments.

**Pull request merged**